haystack_experimental/core/super_component/super_component.py

```diff
@@ -0,0 +1,420 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Optional, Tuple
+
+from haystack import Pipeline, component
+from haystack.core.serialization import default_from_dict, default_to_dict
+
+from haystack_experimental.core.super_component.utils import _delegate_default, is_compatible
+
+
+class InvalidMappingError(Exception):
+    """Raised when input or output mappings are invalid or type conflicts are found."""
+
+    pass
+
+
+@component
+class SuperComponent:
+    """
+        A class for creating super components that wrap around a Pipeline.
+
+        This wrapper allows for remapping of input and output socket names between the wrapped
+        pipeline and the external interface. It handles type checking and verification of all
+        mappings.
+
+        :param pipeline: The pipeline wrapped by the component
+        :param input_mapping: Mapping from wrapper input names to lists of pipeline socket paths
+            in format "component_name.socket_name"
+        :param output_mapping: Mapping from pipeline socket paths to wrapper output names
+        :raises InvalidMappingError: If any input or output mappings are invalid or if type
+            conflicts are detected
+        :raises ValueError: If no pipeline is provided
+        """
+
+    def __init__(
+            self,
+            pipeline: Pipeline,
+            input_mapping: Optional[Dict[str, List[str]]] = None,
+            output_mapping: Optional[Dict[str, str]] = None,
+    ) -> None:
+        """
+        Initialize the component with optional I/O mappings.
+
+        :param pipeline: The pipeline to wrap
+        :param input_mapping: Optional input name mapping configuration
+        :param output_mapping: Optional output name mapping configuration
+        """
+        self.pipeline: Pipeline = pipeline
+        self.input_mapping: Optional[Dict[str, List[str]]] = input_mapping
+        self.output_mapping: Optional[Dict[str, str]] = output_mapping
+        self._set_io_types()
+        self.warmed_up = False
+
+    def warm_up(self) -> None:
+        """
+        Warms up the pipeline if it has not been warmed up before.
+        """
+        if not self.warmed_up:
+            self.pipeline.warm_up()
+            self.warmed_up = True
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Convert the pipeline wrapper to a dictionary representation.
```

Comment:
- let's explain here in docstrings that if you want to create a custom SuperComponent with custom attributes, you MUST appropriately override this method
- nit: let's use super component instead of pipeline wrapper